### ISSUE TITLE
[Feature] Option to fetch history rows with undefined keys

### ISSUE THREAD

**Issue by karolpiczak** [wandb/client#2266]
I don't see an efficient way to get a history object for a number of runs where the desired keys might not be available in some of the runs.

One option is to fetch all keys and filter locally, which is especially impractical for media types which I don't need in many situations. An alternative approach requires an accurate knowledge of run/key mapping and building from smaller queries. Unfortunately, I have not found an easy way to get this kind of information through the API.

I would like to have an option to fetch a filtered view (through `history` and `scan_history`) without discarding rows with missing values. Or at least an option to `ignore_keys` so that at least some of the columns can be excluded before being sent.

**Comment by cvphelps** [comment]
@karolpiczak you're right, currently we only return rows with _all_ instead of _any_ of the keys present. Thank you for suggesting this improvement. We're not going to address this soon, but we'll keep in our backlog.

**Comment by LucasAlegre** [comment]
Hi, is there any update on this? I am unable to retrieve my rows because they have at least one missing column each.

**Comment by jacobromero** [comment]
Hi @karolpiczak sorry for the really late reply, unfortunately we did not have the bandwidth to address this issue. 
I am taking a look at this now to hopefully see if we can get this resolved for you.

If I am understanding correctly, you are trying to get the history for your runs if they contain a specific set of keys?

for a given run you can check which history keys are available with something like the following
```python
import wandb

api = wandb.Api()

run = api.run("{entity}/{project}/{run_id}")
run.history_keys["keys"].keys()
```
